test/connectors/CustomsDataStoreConnectorSpec.scala

```diff
@@ -150,10 +150,68 @@ class CustomsDataStoreConnectorSpec extends SpecBase {
     }
   }
 
+  "retrieveUnverifiedEmail" must {
+    "return EmailUnverifiedResponse with unverified email value" in new Setup {
+
+      when(requestBuilder.execute(any[HttpReads[EmailUnverifiedResponse]], any[ExecutionContext]))
+        .thenReturn(Future.successful(emailUnverifiedRes))
+
+      when(mockHttpClient.get(any[URL]())(any())).thenReturn(requestBuilder)
+
+      running(app) {
+        val result = await(customsDataStoreConnector.retrieveUnverifiedEmail)
+        result mustBe emailUnverifiedRes
+      }
+    }
+
+    "return EmailUnverifiedResponse with None for unverified email if there is an error while" +
+      " fetching response from api" in new Setup {
+
+      when(requestBuilder.execute(any[HttpReads[EmailUnverifiedResponse]], any[ExecutionContext]))
+        .thenReturn(Future.failed(new RuntimeException("error occurred")))
+
+      when(mockHttpClient.get(any())(any())).thenReturn(requestBuilder)
+
+      running(app) {
+        val result = await(customsDataStoreConnector.retrieveUnverifiedEmail)
+        result.unVerifiedEmail mustBe empty
+      }
+    }
+  }
+
+  "verifiedEmail" must {
+    "return verified email when email-display api call is successful" in new Setup {
+
+      when(requestBuilder.execute(any[HttpReads[EmailVerifiedResponse]], any[ExecutionContext]))
+        .thenReturn(Future.successful(emailVerifiedRes))
+
+      when(mockHttpClient.get(any())(any())).thenReturn(requestBuilder)
+
+      running(app) {
+        val result = await(customsDataStoreConnector.verifiedEmail)
+        result mustBe emailVerifiedRes
+      }
+    }
+
+    "return none for verified email when exception occurs while calling email-display api" in new Setup {
+
+      when(requestBuilder.execute(any[HttpReads[EmailVerifiedResponse]], any[ExecutionContext]))
+        .thenReturn(Future.failed(UpstreamErrorResponse("error occurred", 500, 500)))
```

Comment:
Please use INTERNAL_SERVER_ERROR instead of 500 to avoid magic number warnings and to make it more readable